lib/vfs/src/tmp_fs.rs

```diff
@@ -0,0 +1,79 @@
+#![allow(dead_code)]
+#![allow(unused)]
+use std::collections::HashMap;
+use std::io::prelude::*;
+use std::io::SeekFrom;
+use std::io::{self};
+use std::path::{Path, PathBuf};
+use std::result::Result as StdResult;
+use std::sync::atomic::AtomicU32;
+use std::sync::Arc;
+use std::sync::Mutex;
+#[allow(unused_imports, dead_code)]
+use tracing::{debug, error, info, trace, warn};
+
+use crate::mem_fs;
+use crate::Result as FsResult;
+use crate::*;
+
+#[derive(Debug, Default, Clone)]
+pub struct TmpFileSystem {
```

Comment:
This just forwards all function calls to the inner mem fs, there is no custom behaviour, so it still seems useless?

> //! Wraps the memory file system implementation - this has been
> //! enhanced to support mounting file systems, shared static files,
> //! readonly files, etc...

I don't quite get this comment, since this is not functionality added by the tmp fs?

Reply:
@theduke It has probably been extended by John on some other branch. I removed TmpFs, but since John said it was used by the wasmer.sh shell, I've readded it.

I don't know what to do here, you'd need to ask @john-sharratt. It's very likely that there is missing functionality that we can port in a second PR. For now I'd keep it in even though it seems useless.

Reply:
Sounds good, we'll need to do a lot of cleanup during/after the wasix merge anyway.